open-grant-proposals/orbitdb-pinning/README.md

```diff
@@ -0,0 +1,194 @@
+# Open Grant Proposal: `OrbitDB Filecoin Integration`
+
+**Name of Project:** `OrbitDB Filecoin Integration`
+
+**Proposal Category:** `app-dev`
+
+**Proposer:** `@aphelionz` on behalf of `@eqlabs`
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and
+APACHE2 licenses?:** Yes
+
+# Project Description
+
+[OrbitDB](https://github.com/orbitdb/orbit-db) is a distributed database engine that stores its
+data on IPFS as Conflict-free Replicated Data Types (CRDTs). OrbitDB has a number of prominent
+users including [Metamask](https://metamask.io), [bitcoin.com](https://bitcoin.com),
+[3Box](https://3box.io/), [AvionDB](https://dbdb.io/db/aviondb),
+and [BitcoinCash](https://www.bitcoincash.org/).
+
+Of all the features and improvements requested by our community, reliable pinning is perhaps the
+most common. An OrbitDB user wants their data to persist in the distributed swarm safely and
+reliably, with minimal configuration and attention required.
```

Comment:
Can you describe what the interface would look like to orbit_db users to persist their data to filecoin? How integrated would this be into the normal configuration of orbitdb?

Reply:
Sure. There are two potential places in the OrbitDB workflow where filecoin integration can be injected.

1. At the creation of an OrbitDB instance, which has multiple databases

```JavaScript
// IPFS is currently passed in here, along with options. 
const orbitdb = OrbitDB.createInstance(ipfs, options) 
```

Filecoin configuration here could be composed in either as another argument i.e. `createInstance(ipfs, filecoin, options)`, or simply as part of the `options` argument.

2. Similarly, once an OrbitDB instance is created, more configuration can happen at the individual store level

```JavaScript
// Options or an addtional argument can be used here similar to above
const docsDb = await orbitdb.docstore(options)
```

Whether it's one or the other or both is TBD, as I think there are good reasons for either situation.